File: frontend/src/types/message.ts
// Message domain types.
// Kept in sync with backend/app/schemas/message.py.

export interface Message {
  id: number
  sender_id: number
  recipient_id: number
  job_id: number | null
  body: string
  read: boolean
  created_at: string
}

export interface ThreadSummary {
  user_id: number
  user_name: string
  last_message_body: string
  last_message_at: string
  unread_count: number
}

File: backend/app/schemas/message.py
"""Message request/response schemas."""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class MessageCreate(BaseModel):
    """Fields required to send a message."""

    recipient_id: int
    body: str
    job_id: Optional[int] = None


class MessageResponse(BaseModel):
    """Public-facing message representation."""

    model_config = ConfigDict(from_attributes=True)

    id: int
    sender_id: int
    recipient_id: int
    job_id: Optional[int] = None
    body: str
    read: bool
    created_at: datetime


class ThreadSummary(BaseModel):
    """One row per counterpart a user has exchanged messages with."""

    user_id: int
    user_name: str
    last_message_body: str
    last_message_at: datetime
    unread_count: int
